src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/ICustomMarshalerTests.cs

```diff
@@ -0,0 +1,537 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class ICustomMarshalerTests
+    {
+        // To avoid having to create a native test library to reference in tests that
```

Comment:
@hughbe 
Take a look at https://github.com/dotnet/coreclr/issues/9287 , it has a test attached to it. I was not merged since it needed the bug fixed S.R.InteropServices from CoreFX